src/js/selection/SelectionManager.js

```diff
@@ -101,14 +107,36 @@
 
     this.pastePixels_(frame, pixels);
 
-    $.publish(Events.PISKEL_SAVE_STATE, {
-      type : pskl.service.HistoryService.REPLAY,
-      scope : this,
-      replay : {
-        type : SELECTION_REPLAY.PASTE,
-        pixels : JSON.parse(JSON.stringify(pixels.slice(0)))
+    // Offset the pasted selection from the original location.
+    var tool = pskl.app.drawingController.currentToolBehavior;
+    var isSelectionTool = tool instanceof pskl.tools.drawing.selection.BaseSelect;
+    if (isSelectionTool) {
```

Comment:
Open question: Should this become an early-out?  Should we do _anything_ in this method if we don't have a selection tool active?